src/main/java/com/devonfw/ide/sonarqube/common/impl/check/DevonUcImplSecurityConstraintCheck.java

```diff
@@ -0,0 +1,159 @@
+                + "Please use one of the following annotations on Use-Case implementation methods: "
+                + REQUIRED_ANNOTATIONS.toString());
+      }
+    }
+  }
+
+  /**
+   * @param context Context of analysis containing the parsed tree.
+   * @return Class tree.
+   */
+  protected ClassTree getClassTree(JavaFileScannerContext context) {
+
+    CompilationUnitTree parsedTree = context.getTree();
+    List<Tree> types = parsedTree.types();
+
+    for (Tree tree : types) {
+      if (tree instanceof ClassTree) {
+        return (ClassTree) tree;
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Returns the Uc interface implemented by the checked class, if there is one.
+   *
+   * @param tree Tree currently being investigated.
+   * @return TypeTree instance of the Uc interface.
+   */
+  protected TypeTree getUcInterface(ClassTree tree) {
+
+    List<TypeTree> interfaces = tree.superInterfaces();
+    String className = tree.simpleName().name();
+    Pattern interfaceRegEx = Pattern.compile(className.replaceAll("Impl", ""));
+
+    if (interfaces.isEmpty()) {
+      return null;
+    }
+
+    for (TypeTree interfaceTree : interfaces) {
+      if (interfaceRegEx.matcher(interfaceTree.toString()).matches()) {
+        return interfaceTree;
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Returns all methods of the given tree.
+   *
+   * @param tree Tree currently being investigated.
+   * @return List of MethodTree.
+   */
+  protected List<MethodTree> getMethodsOfTree(ClassTree tree) {
+
+    List<Tree> membersOfTree = tree.members();
+    List<MethodTree> methodsOfTree = new ArrayList<>();
+
+    for (Tree member : membersOfTree) {
+      if (member.is(Tree.Kind.METHOD)) {
+        methodsOfTree.add((MethodTree) member);
+      }
+    }
+
+    return methodsOfTree;
+  }
+
+  private boolean isMethodProperlyAnnotated(MethodTree method) {
+
+    List<AnnotationTree> annotationsOfMethod = method.modifiers().annotations();
+    boolean hasOverrideAnnotation = false;
+    boolean hasRequiredAnnotation = false;
+
+    for (AnnotationTree annotation : annotationsOfMethod) {
+      if (annotation.annotationType().toString().equals("Override")) {
+        hasOverrideAnnotation = true;
+      }
+
+      if (REQUIRED_ANNOTATIONS.contains(annotation.annotationType().toString())) {
+        hasRequiredAnnotation = true;
+      }
+    }
+
+    if ((!hasOverrideAnnotation) || (hasOverrideAnnotation && hasRequiredAnnotation)) {
+      return true;
+    } else {
+      return false;
+    }
+
+  }
+
+  /**
+   * Called in case of a dependency that is devonfw compliant.
+   *
+   * @param source the {@link JavaType} to analyze.
+   * @param target the {@link JavaType} used by the source type (as dependency).
+   * @return the message of an issue to create due to an undesired dependency or {@code null} if dependency is fine.
+   */
+  @Override
+  protected String checkDependency(JavaType source, JavaType target) {
```

Comment:
If you override a method, the JavaDoc is inherited automatically from super-class/-interface. Please do not repeat it here (simply remove this JavaDoc).